scipy/linalg/decomp_cholesky.py

```diff
@@ -2,26 +2,33 @@
 
 from __future__ import division, print_function, absolute_import
 
-from numpy import asarray_chkfinite, asarray
+from numpy import asarray_chkfinite, asarray, atleast_2d
 
 # Local imports
 from .misc import LinAlgError, _datacopied
 from .lapack import get_lapack_funcs
 
 __all__ = ['cholesky', 'cho_factor', 'cho_solve', 'cholesky_banded',
-            'cho_solve_banded']
+           'cho_solve_banded']
 
 
 def _cholesky(a, lower=False, overwrite_a=False, clean=True,
-                check_finite=True):
+              check_finite=True):
     """Common code for cholesky() and cho_factor()."""
 
-    if check_finite:
-        a1 = asarray_chkfinite(a)
-    else:
-        a1 = asarray(a)
-    if len(a1.shape) != 2 or a1.shape[0] != a1.shape[1]:
-        raise ValueError('expected square matrix')
+    a1 = asarray_chkfinite(a) if check_finite else asarray(a)
+
+    # Quick return for empty arrays
+    if a1.size == 0:
+        return a, lower
```

Comment:
My rationale is that if an empty array is provided I don't want to give a wrong error message about its shape or dimension. Instead just want to return back asap. such that nobody bothers to make a 2D empty array just to get a pass. 

But now I noticed that it should return `a1`  instead but not `a`.